packages/itwinui-css/src/button/button.scss

```diff
@@ -22,49 +19,99 @@ $iui-button-padding-large: $iui-xs * 6;
   position: relative;
   border-radius: $iui-border-radius;
   line-height: $iui-line-height;
-  box-shadow: none;
   font-size: $iui-font-size;
   font-weight: $iui-font-weight-normal;
-  text-decoration: none;
   user-select: none;
   cursor: pointer;
   white-space: nowrap;
-  border: 1px solid transparent;
+  border: 1px solid var(--_iui-button-border-color);
+  background-color: var(--_iui-button-background-color);
   color: var(--_iui-button-text-color);
-  @media (prefers-reduced-motion: no-preference) {
-    transition: color $iui-speed-fast ease-out, background-color $iui-speed-fast ease-out,
-      border-color $iui-speed-fast ease-out;
+  transition: background-color $iui-speed-fast ease-out, border-color $iui-speed-fast ease-out;
+  -webkit-tap-highlight-color: transparent;
+
+  @at-root a#{&} {
+    text-decoration: none;
   }
 
-  &:hover {
-    --_iui-button-text-color: rgba(var(--iui-color-foreground-body-rgb), var(--iui-opacity-1));
+  // #region Sizing
+  @include iui-button-size;
 
-    text-decoration: none;
+  &[data-iui-button-size='small'] {
+    @include iui-button-size(small);
   }
 
-  @include iui-focus;
+  &[data-iui-button-size='large'] {
+    @include iui-button-size(large);
+  }
+  // #endregion Sizing
+
+  &[aria-disabled='true'] {
+    @include iui-focus;
```

Comment:
Shouldn't `iui-focus` be included for all states rather than _only_ for disabled?

Reply:
`iui-focus` is set on the variant types due to differences between the variants:
- Default button = 2px outline
![Screen Shot 2022-07-08 at 12 28 30 PM](https://user-images.githubusercontent.com/849817/178032763-ac64b7e9-22c5-4857-8e82-00b64fb2d8ba.png)
- Borderless button = 1px outline
![Screen Shot 2022-07-08 at 12 28 48 PM](https://user-images.githubusercontent.com/849817/178032810-e1b8c717-8c11-427c-973f-24a121f7563f.png)
- CTA & High visibility button = 1px outline, -1px inset
![Screen Shot 2022-07-08 at 12 29 14 PM](https://user-images.githubusercontent.com/849817/178032861-15584dd7-8431-45dc-a141-8721a42b9a53.png) ![Screen Shot 2022-07-08 at 12 29 26 PM](https://user-images.githubusercontent.com/849817/178032888-e6dd7569-3b00-48c3-8d40-94f74ed90723.png)



Then for all `aria-disabled="true"` buttons, no matter the variant type, we reset the `iui-focus` to 1px outline.
![Screen Shot 2022-07-08 at 12 30 13 PM](https://user-images.githubusercontent.com/849817/178032991-eeb5de06-ad03-4516-9a65-af8af182826d.png)